README.md

```diff
@@ -145,7 +174,8 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+npm test
+This should run the code. The code is in the lang.js file within the test folder.
```

Comment:
The exercise asks you to translate the table into code. The intent there was to actually have a lookup table, rather than a bunch of if statements. I'm dropping two points for that.

Also, no deduction, but keep an eye of for duplication in your tests. Here's my version of your code:

~~~ javascript
var mocha    = require('mocha');
var describe = mocha.describe;
var it       = mocha.it;
var assert   = require('assert');

let END = 99;

states = {
  0:  { "the":    1 },
  1:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  2:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  3:  { "ate":    4, "ran":     4 },
  4:  { "slowly": 4, "noisily": 4, "EOI": END },
  5:  { "EOI":    END }
};


let getNextState = (currentState, nextWord) =>  {
    let transitions = states[currentState];
    let nextState;

    if (transitions && (nextState = transitions[nextWord]))
		return nextState;
	else
		return -1;
}

function stateMachine(sentence) {
    return sentence.reduce(getNextState, 0);
}

function testSentence(sentence) {
    let result = stateMachine(sentence.split(" "));
    return result == END;
}

describe('Sentence Builder', () => {
    let goodSentences = [
        "the dog ate EOI",
        "the cat ran EOI",
        "the lazy dog ate EOI",
        "the smelly cat ran EOI",
        "the lazy smelly dog ran EOI",
        "the smelly lazy cat ate EOI",
        "the cat ate slowly EOI",
        "the dog ran noisily EOI",
        "the smelly cat ate noisily EOI",
        "the lazy dog ran noisily slowly EOI",
        "the smelly lazy dog ran slowly noisily EOI"
    ]

    let badSentences = [
        "the cool dog ate slowly EOI",
        "what the cool dog did EOI",
        "this sentence is all kinds of wrong EOI"
    ];

    it('sentences that should be found', () => {
        for (let sentence of goodSentences) {
            assert(testSentence(sentence), sentence);
        }
    })
    it('sentences that should not be found', () => {
        for (let sentence of badSentences) {
            assert(!testSentence(sentence), sentence);
        }
    })
});
~~~

# -2